Statistics/Distributions/Empirical.cs

```diff
@@ -420,29 +403,48 @@ public bool Equals(IDistribution distribution)
 
         public double InverseCDF(double p)
         {
-            int index = Array.BinarySearch(CumulativeProbabilities,p);
-            if (index >= 0)
+            if (Truncated && _Constructed)
             {
-                return ObservationValues[index];
+                p = _ProbabilityRange.Min + (p) * (_ProbabilityRange.Max - _ProbabilityRange.Min);
```

Comment:
looks good but i never trust order of operations. i only trust parenthetical statements (interpret that as you would like)!